code/game/objects/effects/effect_system/smoke.dm

```diff
@@ -150,6 +153,8 @@
 
 //proc to check if smoke can expand to another turf
 /obj/effect/particle_effect/smoke/proc/check_airblock(turf/T)
+	if(!T)
```

Comment:
This proc is called by `spread_smoke()` in where `get_adjacent_open_turfs()` already filters invalid values. There is no need for this check.